intelmq/lib/utils.py

```diff
@@ -808,3 +812,51 @@ def file_name_from_response(response: requests.Response) -> str:
     except KeyError:
         file_name = response.url.split("/")[-1]
     return file_name
+
+
+def list_all_bots() -> str:
+    mod = importlib.import_module('intelmq.bots')
+    dirs = ['collectors', 'parsers', 'experts', 'outputs']
+    allowed_types = ['bool', 'NoneType', 'str', 'int']
+    bot_types = [
+        'CollectorBot',
+        'ParserBot',
+        'OutputBot',
+        'Bot'
+    ]
+    root = mod.__path__[0]
+    root_dict = {}
+
+    for d in dirs:
+        root_dict[d.capitalize()] = {}
+        for dirs in os.listdir(os.path.join(root, d)):
+            bot = Path(os.path.join(root, d, dirs))
```

Comment:
You're mixing `pathlib` and `os` at will, you can simple concatenate `pathlib` paths with `/` and iterate over directories using `iterdir`.
You could use something like this to get all the python files in the bot directories:
```
botfiles = [botfile for botfile in pathlib.Path('intelmq/bots').glob('**/*.py') if botfile.is_file() and botfile.name != '__init__.py']
```

Then you have a flat list of python files as `Path` objects and you don't have to use multiple nested for loops, which makes the code way more readable